draft-parecki-oauth-first-party-native-apps.md

```diff
@@ -443,6 +443,12 @@ In a traditional OAuth deployment using the redirect-based authorization code fl
 Because of these risks, the authorization server MAY decide to require that the user go through a redirect-based flow at any stage of the process based on its own risk assessment.
 
 
+## Credential Attacks {#credential-attacks}
+
+The nature of moving from browser-based authentication to a backend API which is capable of directly receiving user credentials exposes a risk to attacks such as credential stuffing and brute forcing credentails to see if an authorization code is returned by the authorization challenge endpoint. An authorization server may already have a combination of built-in or 3rd party security tools in place to monitor and reduce this risk in the browser-based authentication.  
```

Comment:
It is critical to remember that the Authorization Server should not allow a client to leverage this path unless the AS is very confident the client is a first party application (i.e. "owned" by the same entity as the authorization server). Ensuring client_authentication is available on the API is important. I'd like to discuss this in an issue before merging the PR.